src/transformStateWithClones.js

```
@@ -7,7 +7,37 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const actionHistory = [];
+  const copyState = { ...state };
+
+  for (const el of actions) {
+    const type = el.type;
+
+    if (type === 'clear') {
+      for (const key in copyState) {
+        delete copyState[key];
+      }
+      actionHistory.push({ ...copyState });
+    }
+
+    if (type === 'addProperties') {
+      const addedProp = Object.assign(copyState, el.extraData);
+
+      actionHistory.push({ ...addedProp });
+    }
+
+    if (type === 'removeProperties') {
+      for (const rem of el.keysToRemove) {
+        delete copyState[rem];
+      }
+
+      const removeProp = { ...copyState };
+
+      actionHistory.push(removeProp);
+    }
```

There is no `default`/fallback branch for unknown action types. The checklist requires a default case for error handling. This violates checklist item #3: "3. [CODE STYLE]: switch/case should always have default case for error handling." Add a `default` case (throw an Error or otherwise make behavior explicit) so unsupported action types are not silently ignored.